addons/sale_purchase/models/sale_order_line.py

```diff
@@ -229,6 +229,7 @@ def _purchase_service_match_purchase_order(self, partner, company=False):
             ('partner_id', '=', partner.id),
             ('state', '=', 'draft'),
             ('company_id', '=', (company and company or self.env.company).id),
+            ('order_line.sale_order_id', '=', self.order_id.id)
```

Comment:
This line could introduce a performance issue, I think if we have a lot of POLs in db. 🤔 

We could maybe do a read_group on purchase.order.line instead.

Something like:

```py
    def _purchase_service_match_purchase_order(self, partner, company=False):
        purchase_order_line_read_group = self.env['purchase.order.line']._read_group(
            [
                ('partner_id', '=', partner.id),
                ('state', '=', 'draft'),
                ('company_id', '=', (company and company or self.env.company).id),
                ('order_line.sale_order_id', '=', self.order_id.id),
            ],
            ['order_id'],
            order='order_id',
        )
        return self.env['purchase.order'].browse([
            purchase_order.id
            for purchase_order, in purchase_order_line_read_group
        ])
```

However, I am a bit surprised we don't have `limit=1` in that search since the result used in other method assumes the result contains only one purchase_order. 🤯 

I guess we have to add `limit=1` to be sure to return only one record but I am not sure, I will let Sales team check that.

If it is needed then I think we could maybe do something like that:

```py
    def _purchase_service_match_purchase_order(self, partner, company=False):
        return self.env['purchase.order.line']._read_group(
            [
                ('partner_id', '=', partner.id),
                ('state', '=', 'draft'),
                ('company_id', '=', (company and company or self.env.company).id),
                ('sale_order_id', '=', self.order_id.id),
            ],
            ['order_id'],
            order='order_id',
            limit=1,
        )[0][0]
```